test/integration/reasoner/ResolutionTest.java

```diff
@@ -0,0 +1,451 @@
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.commit();
+            }
+        }
+        long ruleTraversalAnswerCount = 3L;
+        long atomic1TraversalAnswerCount = 3L;
+        long conjunctionTraversalAnswerCount = 0L;
+        // the recursively produced answers will be identical, so will be deduplicated
+        long answerCount = conjunctionTraversalAnswerCount + atomic1TraversalAnswerCount + ruleTraversalAnswerCount + atomic1TraversalAnswerCount - atomic1TraversalAnswerCount;
+        Conjunction conjunctionPattern = parseConjunction("{ " + atomic1 + " }");
+        createRootAndAssertResponses(conjunctionPattern, answerCount);
+    }
+
+    @Ignore // TODO Un-ignore, ignored until rules are ready to use
+    @Test
+    public void answerRecorderTest() throws InterruptedException {
+
+        String atomic1 = "$p1 isa person, has name \"Bob\";";
+        String atomic2 = "$p1 isa person, has age 42;";
+        String rule = "rule bobs-are-42: when { $p1 has name \"Bob\" } then { $p1 has age 42; };";
+        try (RocksSession session = schemaSession()) {
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().define(Graql.parseQuery(
+                        "define person sub entity, owns age, owns name;" +
+                                "age sub attribute, value long;" +
+                                "name sub attribute, value string;" +
+                                rule));
+                transaction.commit();
+            }
+        }
+        try (RocksSession session = dataSession()) {
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.commit();
+            }
+        }
+
+        long atomic1TraversalAnswerCount = 3L;
+        long ruleTraversalAnswerCount = 0L;
+        long atomic2TraversalAnswerCount = 0L;
+        long conjunctionTraversalAnswerCount = 3L;
+        long answerCount = conjunctionTraversalAnswerCount + atomic2TraversalAnswerCount + ruleTraversalAnswerCount + atomic1TraversalAnswerCount;
+        Conjunction conjunctionPattern = parseConjunction("{ " + atomic2 + " }");
+
+        LinkedBlockingQueue<ResolutionAnswer> responses = new LinkedBlockingQueue<>();
+        AtomicLong doneReceived = new AtomicLong(0L);
+        EventLoopGroup elg = new EventLoopGroup(1, "reasoning-elg");
```

Comment:
i think even though this test is possibly going to have to change a fair amount, we should update it to be similar style to the others - eg. get `ResolverRegistry` via the `Reasoner`, etc.

Reply:
I missed that one!